src/utils.tsx

```diff
@@ -0,0 +1,8 @@
+export function stringToEnum(str: string, enumType: any): any {
+  let matchingKeys = Object.keys(enumType).filter((key) => enumType[key] === str);
+  if (matchingKeys.length === 0) {
+    throw('Invalid value "' + str + '" given for enum with values [' + 
+      Object.keys(enumType).map(key => enumType[key]).join(', ') + ']');
```

Comment:
1. String interpolation might simplify the code a bit. 2. I believe throwing `new Error(string)` is preferred over throwing a string because the Error contains additional information which can be useful when debugging, like stack trace.